### PR TITLE
Fix CMake enable/disable logic for Fortran API

### PR DESCRIPTION
**Pull request by dalg24**
See [TriBITS](https://tribits.org/doc/TribitsDevelopersGuide.html#how-to-check-for-and-tweak-tribits-enable-cache-variables) manual for reference.

### CODE REVIEW

**Comment by dalg24** [comment]
There is no test that demonstrate that this resolves the issue but here is how to check that it does the trick using our dev container:
```bash
[host]$ cd docker
[host]$ docker-compose up -d
[host]$ docker-compose exec dtk_dev bash
[container]# cd $TRILINOS_DIR/DataTransferKit
[container]# [ ! -d build ] && mkdir build
[container]# cd build
[container]# ../scripts/docker_cmake -D Trilinos_ENABLE_Fortran=OFF
```
Look toward the end of what was printed to the screen and search for:
```
Processing enabled package: DataTransferKit (Utils, Kokkos, Interface, BasicGeometryAdapters, IntrepidAdapters, Operators, C_API, STKMeshAdapters, MoabAdapters, LibmeshAdapters, ClassicDTKAdapters, Tests, Examples)
```
Make sure that `Fortran_API` is not listed any more.

**Comment by aprokop** [comment]
So why did it not work before?

**Comment by dalg24** [comment]
Variable `Trilinos_ENABLE_DataTransferKitFortran_API` was set to `FALSE` **after** the graph of dependencies was computed.

**Comment by aprokop** [comment]
Thanks, could you please amend the commit to include this information? Or, create an issue with it that this commit fixes? I think it would be useful to know in the future.

**Comment by dalg24** [comment]
Per @aprokop's comment, created issue #192 and amended commit message.

File: cmake/RepositoryDependenciesSetup.cmake [hunk 1]
```diff
@@ -1 +1,11 @@
 SET_DEFAULT(DataTransferKit_REPOSITORY_MASTER_EMAIL_ADDRESS coupler-infrastructure@casl-dev.ornl.gov)
+##---------------------------------------------------------------------------##
+## Disable Fortran_API if Trilinos_ENABLE_Fortran is FALSE
+##---------------------------------------------------------------------------##
+IF (NOT ${PROJECT_NAME}_ENABLE_Fortran)
+    IF ("${${PROJECT_NAME}_ENABLE_DataTransferKitFortran_API}" STREQUAL "")
+        MESSAGE("-- " "NOTE: Setting ${PROJECT_NAME}_ENABLE_DataTransferKitFortran_API=OFF because"
+                "${PROJECT_NAME}_ENABLE_Fortran = '${DataTransferKit_ENABLE_Fortran}'")
```
Comment (Rombur) [inline cmake/RepositoryDependenciesSetup.cmake]: Use `${PROJECT_NAME}` everywhere or nowhere. Why did you replace only one of the `${PROJECT_NAME}` by `DataTransferKit`?

Comment (dalg24) [inline cmake/RepositoryDependenciesSetup.cmake]: The variable `PROJECT_NAME` actually refers to Trilinos.  We usually use `PACKAGE_NAME` for DTK but it is not defined at the time the dependency graph gets processed.

Comment (Rombur) [inline cmake/RepositoryDependenciesSetup.cmake]: So you have replaced `${Trilinos_ENABLE_Fortran}` by `${DataTransferKit_ENABLE_Fortran}`. Does such a variable exist?

Comment (Rombur) [inline cmake/RepositoryDependenciesSetup.cmake]: Just to be clear, I am talking about the message not the logic.

Comment (dalg24) [inline cmake/RepositoryDependenciesSetup.cmake]: OK.  `${DataTransferKit_ENABLE_Fortran}` was a mistake on my part.  It was supposed to be `${${PROJECT_NAME}_ENABLE_Fortran}`.  Good catch.

**Comment by Rombur** [comment]
@dalg24 please squash and merge yourself.

**Pull request merged**